docs/tutorials/elemental_tutorial.md

```diff
@@ -0,0 +1,373 @@
+
+```bash
+$ oc delete pod mypod
+```
+
+Consequently, the pod should disapper from the OpenShift web console, but let's keep this one running for now.
+
+----
+
+### Resources
+
+Typically one allocates *resources* to containers, but in these examples we refrain from doing that for the sake of brevity. The same pod as above with memory and cpu resources of 200MB...1GB and 0.2CPU...1CPU would read as:
+
+*`pod.yaml`*:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mypod
+  labels:
+    app: serveapp
+    pool: servepod
+spec:
+  containers:
+  - name: serve-cont
+    image: "docker-registry.default.svc:5000/openshift/httpd"
+    resources:
+      requests:
+        memory: "200M"
+        cpu: "200m"
+      limits:
+        memory: "1G"
+        cpu: "1"
+```
+
+----
+
+## Service
+
+Service objects provide consistent network identity to pods.
+
+*`service.yaml`*
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: serve
+  labels:
+    app: serveapp
+spec:
+  ports:
+  - name: 8080-tcp
+    port: 8080
+    protocol: TCP
+  selector:
+    pool: servepod
+```
+
+This service will redirect TCP traffic internally in the project to the pods having labels listed in `spec.selector`. In this case, the service will redirect to all the pods having label `pool: servepod`. If there are multiple pods matching `spec.selector` then traffic is split between pods in a defined way.
+
+Lets try that the service actually works by launching remote shell at container of `mypod`
+
+```bash
+$ oc rsh mypod
+sh-4.2$ ping serve
+PING serve.my-project-with-unique-name.svc.cluster.local (172.30.39.82) 56(84) bytes of data.
+```
+
+## Route
+
+*`route.yaml`*:
+```yaml
+apiVersion: v1
+kind: Route
+metadata:
+  labels:
+    app: serveapp
+  name: myservice
+  annotations:
+    haproxy.router.openshift.io/ip_whitelist: 192.168.1.0/24 10.0.0.1
+spec:
+  host: myservice-my-project-with-unique-name.rahtiapp.fi
+  to:
+    kind: Service
+    name: serve
+    weight: 100
+```
+
+This route will redirect traffic from internet to that service in the cluster whose `metadata.name` equals `spec.to.name`. 
+
+This particular route also allows traffic only from subnet `192.168.1.0/24` and the IP `10.0.0.1`. Security-wise it is highly encouraged to utilize ip-whitelisting. 
+
+!!! Caution
+    If the whitelist entry is malformed, OpenShift will discard the whitelist and traffic is allowed from everywhere.
+
+* If the service `spec.to.name` has multiple ports defined then one should define `spec.port.targetport`
+* By default the hostname is `metadata.name` + `-` + project name + `.rahtiapp.fi` unless otherwise specified in `spec.host`.
+
+So far we have set up a pod, a service and a route. If the physical server where to pod lives gets shut down one has to manually start the pod again with `oc create -f pod.yaml`. The ReplicationController and ReplicaSet objects are mechanism that will, roughly speaking, do that for the user. 
+
+## ReplicationController
```

Comment:
DeploymentConfigs support triggering via image changes and openshift regularly creates DeploymentConfigs via oc run / oc new-app commands. So I would prefer to treat ReplicationControllers and DeploymentConfigs over Deployments+ReplicaSets even though the latter are more prevalent and recommended in the K8s.

However, exposing ReplicationControllers here might seem to be out of place *or* it can be seen as an introduction to scattered information across Rahti/OpenShift/K8s documentation. Now the reader has rubber floatation devices so we can push them over the cliff to [documentation](https://v1-10.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/). Or, well.. Maybe rather here [here](https://kubernetes.io/docs/home/).

Reply:
I added more explanation about rc vs rs here:

```
The ReplicationControllers are functionally near ReplicaSets, treated
in Chapter "[Kubernetes and OpenShift
Concepts](../introduction/background#kubernetes-and-openshift-concepts)".
In essence, a ReplicationController can be transformed in to a ReplicaSet by
changing `metadata.labels` to `metadata.matchLabels` and setting 
`kind: ReplicaSet`. The motivation to understand the ReplicationController 
object is that [DeploymentConfig](advanced_tutorial#deploymentconfig) 
objects generate ReplicationControllers.
```